doc/sphinx/practical-tools/utilities.rst

```diff
@@ -364,6 +380,28 @@ line timing data:
     .. note::
        This target requires python to build the table.
 
+    .. note::
+       This target follows the same sorting order as the
+       ``print-pretty-timed-diff`` target, and supports the same
+       options for the ``TIMING_SORT_BY`` variable.
+
+    .. note::
+       By default, two lines are only considered the same if the
+       character offsets and initial code strings match.  Passing
+       ``TIMING_FUZZ=N`` relaxes this constraint by allowing the
+       character offsets to differ by up to ``N`` characters, as long
```

Comment:
Is this an N character prefix, suffix or any N characters scattered in the method names?  "abc" to "abcX" or "ac" or "aBc" or "acb" all match with N=1?  Order matters, eg "abc" doesn't match "cba" for N=1?

Now that I think about it just a little, this option doesn't seem very general--it probably solves your problem though probably not really cleanly.

Reply:
I think you missed the word "offsets".  The strings (what we can see of them) must be identical, and the line length must be identical.  Without this option, if you add one space in a comment at the beginning of the file, then every subsequent command is considered completely different due to occurring at different locations in the two versions of the document.
Perhaps the use of "characters" after "`N`" is misleading (but I don't know what else to call the units that I'm shifting the document locations by).  Do you have a suggestion of how to make this clearer?

Reply:
Perhaps my use of "initial code strings" is also confusing; what I meant to point at here is that only a very limited number of characters from each line appear in the log, so we don't actually have access to the full commands generating each timing.

Reply:
Yeah, I missed "offsets".

Dropping "characters" after "N" would make it a little harder to misread.  The first part of the sentence make it clear that the units for the offsets are characters.

Why are the offsets needed?  Are the "initial code strings" not unique?  (And if there are occasionally duplicates, there's no reasonable way to qualify the names?)

The timing/performance documentation needs a couple paragraphs of overview: What files are used to collect data?  What is the "timing log"?  What data appears in the build output vs elsewhere?  It's not simple to infer that from the detail-level info.  The number of Notes suggests the material would benefit from reorganization.

Also I wonder if we should eventually do something more robust/easier to use.

Reply:
Initial code strings are absolutely not unique.  Consider the string `Qed.`, for example, which may show up dozens of times.

Perhaps "character locations" would be clearer than "character offsets"?  I am referring to the numbers in things like `Chars 069 - 162 [Definition~foo0~:=~Eval~comp~i...]` that occur in the output of `coqc -time`.

> What files are used to collect data?

You mean where is the data written to, or where is it taken from?

> What is the "timing log"? 

This is what I'm calling the files that I write timing information to.

> What data appears in the build output vs elsewhere?

Per-file timing data appears in the build output.  Per-line timing data is directed straight to files.

> The number of Notes suggests the material would benefit from reorganization.

Indeed.  I'm not sure how to reorganize it well, though.

> Also I wonder if we should eventually do something more robust/easier to use.

Not sure what you mean by this.  While there are a lot of fiddly options, I think the defaults are pretty good, and have been using them pretty easily for years without much trouble to get timing logs.  What gives you the sense of fragility / difficulty to use?  And what do you have in mind as a better option?

Reply:
```suggestion
       character locations to differ by up to ``N``, as long
```